manager/controllers/package_detail.go

```diff
@@ -0,0 +1,193 @@
+
+	var packageInfo models.Package
+	err := query.Find(&packageInfo).Error
+	if err != nil {
+		LogAndRespNotFound(c, err, "package not found")
+		return
+	}
+
+	advisory, err := getAdvisoryByID(packageInfo.AdvisoryID)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "advisory not found")
+		return
+	}
+
+	summary, err := getPackageStrings(packageInfo.SummaryHash)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "summary not found")
+		return
+	}
+
+	descr, err := getPackageStrings(packageInfo.DescriptionHash)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "description not found")
+		return
+	}
+
+	nevra := packageName + "-" + packageInfo.EVRA
+
+	c.JSON(200, PackageDetailResponse{
+		Data: PackageDetailItem{
+			Attributes: PackageDetailAttributes{
+				Description: descr.Value,
+				Summary:     summary.Value,
+				Name:        packageName,
+				Version:     packageInfo.EVRA,
+				AdvisoryID:  advisory.Name,
+			},
+			ID:   nevra,
+			Type: "package",
+		},
+	})
+}
+
+func packageEvraHandler(c *gin.Context, nevra *utils.Nevra) {
+	if !packageNameIsValid(nevra.Name) {
+		c.JSON(http.StatusBadRequest, utils.ErrorResponse{Error: "invalid package name"})
+		return
+	}
+
+	packageName := nevra.Name
+	evra := nevra.EVRAString()
+
+	query := singlePackageEvraQuery(packageName, evra)
+
+	var packageInfo models.Package
+	err := query.Find(&packageInfo).Error
+	if err != nil {
+		LogAndRespNotFound(c, err, "package not found")
+		return
+	}
+
+	advisory, err := getAdvisoryByID(packageInfo.AdvisoryID)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "advisory not found")
+		return
+	}
+
+	summary, err := getPackageStrings(packageInfo.SummaryHash)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "summary not found")
+		return
+	}
+
+	descr, err := getPackageStrings(packageInfo.DescriptionHash)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "description not found")
+		return
+	}
+
+	c.JSON(200, PackageDetailResponse{
+		Data: PackageDetailItem{
+			Attributes: PackageDetailAttributes{
+				Description: descr.Value,
+				Summary:     summary.Value,
+				Name:        packageName,
+				Version:     packageInfo.EVRA,
+				AdvisoryID:  advisory.Name,
+			},
+			ID:   nevra.String(),
+			Type: "package",
+		},
+	})
+}
+
+// @Summary Show me metadata of selected package
+// @Description Show me metadata of selected package
+// @ID LatestPackage
+// @Security RhIdentity
+// @Accept   json
+// @Produce  json
+// @Param    package_name    path    string   true "package_name - latest, nevra - exact version"
+// @Success 200 {object} PackageDetailResponse
+// @Router /api/patch/v1/packages/{package_name} [get]
+func PackageDetailHandler(c *gin.Context) {
+	parameter := c.Param("package_name")
+	if parameter == "" {
+		c.JSON(http.StatusBadRequest, utils.ErrorResponse{Error: "package_param not found"})
+		return
+	}
+
```

Comment:
To be more correct we should add parameter validation here. Nevra should include only specific characters. We should return "BadRequest" if e.g. % is included. I think only `a-zA-Z0-9\+-\._^~` are allowed for nevra. But of course it's not critical.